zebrad/src/components/inbound.rs

```diff
@@ -12,12 +12,21 @@ use futures::{
 use tokio::sync::oneshot;
 use tower::{buffer::Buffer, util::BoxService, Service, ServiceExt};
 
+use zebra_chain as zc;
+use zebra_consensus as zcon;
 use zebra_network as zn;
 use zebra_network::AddressBook;
 use zebra_state as zs;
 
+mod downloads;
+use downloads::Downloads;
+
 type Outbound = Buffer<BoxService<zn::Request, zn::Response, zn::BoxError>, zn::Request>;
 type State = Buffer<BoxService<zs::Request, zs::Response, zs::BoxError>, zs::Request>;
+type Verifier = Buffer<
+    BoxService<Arc<zc::block::Block>, zc::block::Hash, zcon::chain::VerifyChainError>,
+    Arc<zc::block::Block>,
+>;
```

Comment:
I think this would be cleaner as:
```
Buffer<BoxService<Arc<Block>, block::Hash, BoxError>, Arc<Block>>
```
with the imports changed as above.

Reply:
That looks a lot better however there is a problem when calling it here: https://github.com/ZcashFoundation/zebra/blob/ad568fa1731afe34a9e7e4773f8b8d7905ea7808/zebrad/src/commands/start.rs#L70

```
mismatched types

expected struct `std::boxed::Box`, found enum `zebra_consensus::chain::VerifyChainError`

note: expected struct `tower::buffer::Buffer<tower::util::BoxService<_, _, std::boxed::Box<(dyn std::error::Error + std::marker::Send + std::marker::Sync + 'static)>>, _>`
         found struct `tower::buffer::Buffer<tower::util::BoxService<_, _, zebra_consensus::chain::VerifyChainError>, _>`
```